pkg/apisix/cluster.go

```diff
@@ -315,52 +317,74 @@ func (c *cluster) HasSynced(ctx context.Context) error {
 	}
 }
 
-// syncSchema syncs schema from APISIX.
+// syncSchema syncs schema from APISIX regularly according to the interval.
+func (c *cluster) syncSchema(interval time.Duration) {
+	sigCh := make(chan os.Signal, 1)
+	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
```

Comment:
May not handle the signal notification here.

Reply:
Just do it like `syncCache`.

PS: I know the design of `newCluster` is not so elegant, we may add a context for it, but I'd like to optimize this in another PR in the future.

Reply:
But it is different from `syncCache`, which is executed only once. However, the schema needs to be synced regularly, so where should I close the ticker?

Reply:
cc @tokers
I think syncing the schema is a normal task to prevent the schema from changing after the APISIX upgrade, which requires a regular task to run in the background all the time. So this is not the same as `syncCache`.

Do you have any better way to update the schema, if APISIX DP upgrade?

Reply:
cache sync is not an one-kick event, it will be executed once the leader changing.

Reply:
> I think syncing the schema is a normal task to prevent the schema from changing after the APISIX upgrade, which requires a regular task to run in the background all the time. So this is not the same as syncCache.

This is by no means related to the signal handling stuff. Of course, we can do it periodically. A Golang program should only have one place handling the signals.

Reply:
@fgksgf You may tweak the cluster creation logic, and passing the context to it. Here you can listen to the ctx channel.

Reply:
All right, so this comment is about signal processing.

Reply:
Signal processing can be unified.